### DISCUSSION TITLE
Borg backup: files found but no archive located

### DISCUSSION THREAD

**Discussion by BustedSec** [nextcloud/all-in-one#2571] [Questions] [answered]
I am trying to migrate a nextcloud-aio instance from linode to local storage. I created a VM and installed nextcloud-aio, but I'm having an issue locating the borg backup files. The job runs fine and creates the borg folder, but within that folder there is no archive by the name that is given in the backup log.  I tried rsyncing the entire folder over to the new machine and restoring from it, but it doesn't recognize it as a borg backup

The file structure is as follows: 
config  data  hints.74  index.74  integrity.74  nonce  README

here is the log:
```
initializing repository...
using builtin fallback logging configuration
33 self tests completed in 0.23 seconds
Initializing repository at "/mnt/borgbackup/borg"
Key in "<Repository /mnt/borgbackup/borg>" created.
Keep this key safe. Your data will be inaccessible without it.
check_free_space: few segments, not requiring a full free segment
check_free_space: calculated working space for compact as 0 bytes
check_free_space: required bytes 169138, free bytes 79643910144
security: previous location file /root/.config/borg/security/0ac504928b6b823082416f90bb275993ed059cbb3a070335cc5817745cd37304/location not found
security: manifest timestamp file /root/.config/borg/security/0ac504928b6b823082416f90bb275993ed059cbb3a070335cc5817745cd37304/manifest-timestamp not found
security: determined newest manifest timestamp as 
security: remembering previously unknown repository
security: saving state for 0ac504928b6b823082416f90bb275993ed059cbb3a070335cc5817745cd37304 to /root/.config/borg/security/0ac504928b6b823082416f90bb275993ed059cbb3a070335cc5817745cd37304
security: current location   /mnt/borgbackup/borg
security: key type           5
security: manifest timestamp 2023-05-20T00:56:29.881384
security: repository checks ok, allowing access
Synchronizing chunks cache...
Archives: 0, w/ cached Idx: 0, w/ outdated Idx: 0, w/o cached Idx: 0.
Verified integrity of /mnt/borgbackup/borg/index.1
Done.
security: saving state for 0ac504928b6b823082416f90bb275993ed059cbb3a070335cc5817745cd37304 to /root/.config/borg/security/0ac504928b6b823082416f90bb275993ed059cbb3a070335cc5817745cd37304
security: current location   /mnt/borgbackup/borg
security: key type           5
security: manifest timestamp 2023-05-20T00:56:29.881384

By default repositories initialized with this version will produce security
errors if written to with an older version (up to and including Borg 1.0.8).

If you want to use these older versions, you can disable the check by running:
borg upgrade --disable-tam /mnt/borgbackup/borg

See https://borgbackup.readthedocs.io/en/stable/changes.html#pre-1-0-9-manifest-spoofing-vulnerability for details about the security implications.

IMPORTANT: you will need both KEY AND PASSPHRASE to access this repo!
If you used a repokey mode, the key is stored in the repo, but you should back it up separately.
Use "borg key export" to export the key, optionally in printable format.
Write down the passphrase. Store both at safe place(s).

Repository successfully initialized.
Performing backup...
Starting the backup...
Creating archive at "/mnt/borgbackup/borg::20230520_005631-nextcloud-aio"
------------------------------------------------------------------------------
Repository: /mnt/borgbackup/borg
Archive name: 20230520_005631-nextcloud-aio
Archive fingerprint: 7f70d71f5ec4ab66432bfdb517f791862f019b7e9d30743a6a7068ad52a7f551
Time (start): Sat, 2023-05-20 00:56:31
Time (end):   Sat, 2023-05-20 01:05:22
Duration: 8 minutes 51.23 seconds
Number of files: 74573
Utilization of max. archive size: 0%
------------------------------------------------------------------------------
                       Original size      Compressed size    Deduplicated size
This archive:               39.22 GB             35.90 GB             34.28 GB
All archives:               39.22 GB             35.90 GB             34.28 GB

                       Unique chunks         Total chunks
Chunk index:                   77851                88536
------------------------------------------------------------------------------
Pruning the archives...
------------------------------------------------------------------------------
                       Original size      Compressed size    Deduplicated size
Deleted data:                    0 B                  0 B                  0 B
All archives:               39.22 GB             35.90 GB             34.28 GB

                       Unique chunks         Total chunks
Chunk index:                   77851                88536
------------------------------------------------------------------------------
Compacting the archives...
Backup finished successfully on 20.05.2023 - 01:05:26 (00 hours 08 minutes 55 seconds)
```

**Comment by szaimen** [comment]
Hi, on the new server with new AIO instance, did you instead of entering the domain, entered the backup location and backup password in the first step?

**Reply by BustedSec** [reply]
Yes

**Comment by szaimen** [comment, marked as answer]
> I tried rsyncing the entire folder over to the new machine and restoring from it, but it doesn't recognize it as a borg backup

Can you send me the borgbackup container logs when it tries to restore the backup from this archive in the first step when you try to restore the instance from backup wothout entering a domain?

**Reply by oms9** [reply]
Hey, seeing as the original poster has not been responsive, can I just walk through this with you?

I had NC-AIO running in a VirtualBox VM on a spare W11 and I want to migrate it to running in Docker on W11 directly instead as the VM was crashing and hanging frequently. I created the backup which worked fine (have logs for that) and then I zipped it using -0 and copied it over to the host W11 machine but whenever I try to restore the backup using this path: `/run/desktop/mnt/host/d/nextcloud/backups/backup` I get: 

```
No 'borg' directory in the given backup directory found!
Only the files/folders below have been found in the given directory.
.
..
Please adjust the directory so that the borg archive is positioned in a folder named 'borg' inside the given directory!
```

The weird thing is, if I add dummy files or folders to see if they are detected at all instead of just `.` and `..` they don't show up at all in that error message, could this be a permissions problem?

![image](https://github.com/nextcloud/all-in-one/assets/102495444/0c810a6e-c36a-4caf-872e-4cf7cb42fa68)

**Reply by szaimen** [reply]
Hi, can you check if `/run/desktop/mnt/host/d/Nextcloud/Backups/backup` works for you?

**Reply by oms9** [reply]
Hello, I just tried it and it produced the same "No 'borg' directory error. I initially tried it with exact capitalization and then switched to lowercase but to no avail.

**Reply by oms9** [reply]
> Hello, I just tried it and it produced the same "No 'borg' directory error. I initially tried it with exact capitalization and then switched to lowercase but to no avail.

I used the suggested path and added two dummy files to check if they are visible or not. The contents of my `borg` folder are as follows:
```
Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
d-----         2/27/2023  11:46 AM                data
-a----         5/30/2023  12:52 AM            965 config
-a----         5/30/2023  12:51 PM         210684 hints.1690
-a----         5/30/2023  12:51 PM       17825978 index.1690
-a----         5/30/2023  12:51 PM            190 integrity.1690
-a----         5/30/2023  12:46 PM             16 nonce
-a----         9/27/2022   1:13 AM             73 README
```
and when `ls` is run from WSL2:
`.  ..  README  config  data  hints.1690  index.1690  integrity.1690  nonce`

are these correct for the `borg` folder itself?

![image](https://github.com/nextcloud/all-in-one/assets/102495444/9a5ca8b9-58c6-4c59-8439-6aa28a202110)

**Reply by BustedSec** [reply]
Apologies for lack of response. I ended up blowing away my instance and copying over the ncdata to a new instance then running a command to make it scan for that data. I no longer have access to the old one to post files

**Reply by jumblies** [reply]
Same problem here after moving to a new server.  Borg backup could unpack the archive just fine but nextcloud did not find an archive.  Same log message with only ". and .." found.
